src/linear_samplers/vec_col_one_rand_cyclic.jl

```diff
@@ -0,0 +1,42 @@
+# This file is pat of RLinearAlgebra.jl
+# 1. Specifies type
+# 2. Implements sample function
+# 3. Exports Type
+
+# using Random
+
+"""
+    LinSysVecColOneRandCyclic <: LinSysVecColSelect
+
+A mutable structure with a field to store a cycling order. Randomly specifies a cycling
+order over the equations of a linear system. Once this ordering is specified, the ordering
+is kept fixed.
+
+# Fields
+- `order::Union{Vector{Int64},Nothing}`
+
+Calling `LinSysVecColOneRandCyclic()` defaults to setting `order` to `nothing`. The `sample`
+function will handle the re-initialization of the fields once the system is provided.
+"""
+mutable struct LinSysVecColOneRandCyclic <: LinSysVecColSelect
+    order::Union{Vector{Int64},Nothing}
+end
+LinSysVecColOneRandCyclic() = LinSysVecColOneRandCyclic(nothing)
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysVecColOneRandCyclic,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+    if iter == 1
+        type.order = randperm(length(x))
+    end
+
+    eqn_ind = type.order[mod(iter, 1:length(x))]
+    return A[:, eqn_ind], x[eqn_ind]
```

Comment:
If you look at the only implemented column method ([here](https://github.com/numlinalg/RLinearAlgebra.jl/blob/df5f2a797aa2c8f2e44b5739ffbd913e89604107/src/linear_samplers/vec_col_deterministic_cyclic.jl#L34C21-L34C21)), you will see that three things are returned

1. A search direction. In this case, it is the standard vector corresponding to eqn_ind (your column index)
2. The coefficient matrix A
3.  The residual of the normal system

You can see the update used most clearly in this [paper](https://arxiv.org/pdf/2208.04989.pdf) in Equation 3.7. 

1. $S_k$ is the search direction. In the paper, it is a matrix. Here it is a vector.
2. In the paper, $B$ is any symmetric positive definite matrix. Here it is the identity.

So, if we put together the first two things, we have the update
$$x_{k+1} = x_k - s_k \frac{r_k }{\Vert A s_k \Vert_2^2},$$
where $r_k = A^\intercal (A x_k - b)$.

So the sampler should return:

1. The search direction, $s_k$
2. The coefficient matrix, $A$ (though this does not seem to be necessary)
3. The residual of the normal equations, $r_k = A^\intercal (A x_k - b)$